developer/src/kmc-kmn/test/ng-compiler/lexer.tests.ts

```diff
@@ -0,0 +1,1466 @@
+    });
+    it("can recognise a KEYS token", () => {
+      recogniseToken(TokenTypes.KEYS, 'keys');
+    });
+    it("can recognise a KEYMAN token", () => {
+      recogniseToken(TokenTypes.KEYMAN, '$keyman:');
+    });
+    it("can recognise a KEYMANONLY token", () => {
+      recogniseToken(TokenTypes.KEYMANONLY, '$keymanonly:');
+    });
+    it("can recognise a KEYMANWEB token", () => {
+      recogniseToken(TokenTypes.KEYMANWEB, '$keymanweb:');
+    });
+    it("can recognise a KMFL token", () => {
+      recogniseToken(TokenTypes.KMFL, '$kmfl:');
+    });
+    it("can recognise a WEAVER token", () => {
+      recogniseToken(TokenTypes.WEAVER, '$weaver:');
+    });
+    it("can recognise a LEFT_BR token", () => {
+      recogniseToken(TokenTypes.LEFT_BR, '(');
+    });
+    it("can recognise a RIGHT_BR token", () => {
+      recogniseToken(TokenTypes.RIGHT_BR, ')');
+    });
+    it("can recognise a LEFT_SQ token", () => {
+      recogniseToken(TokenTypes.LEFT_SQ, '[');
+    });
+    it("can recognise a RIGHT_SQ token", () => {
+      recogniseToken(TokenTypes.RIGHT_SQ, ']');
+    });
+    it("can recognise a CHEVRON token", () => {
+      recogniseToken(TokenTypes.CHEVRON, '>');
+    });
+    it("can recognise a PLUS token", () => {
+      recogniseToken(TokenTypes.PLUS, '+');
+    });
+    it("can recognise a COMMA token", () => {
+      recogniseToken(TokenTypes.COMMA, ',');
+    });
+    it("can recognise a NOT_EQUAL token", () => {
+      recogniseToken(TokenTypes.NOT_EQUAL, '!=');
+    });
+    it("can recognise a EQUAL token", () => {
+      recogniseToken(TokenTypes.EQUAL, '=');
+    });
+    it("can recognise a RANGE token", () => {
+      recogniseToken(TokenTypes.RANGE, '..');
+    });
+    // it("can recognise a DOLLAR token", () => {
+    //   recogniseToken(TokenTypes.DOLLAR, '$');
+    // });
+    it("can recognise a U_CHAR token", () => {
+      recogniseToken(TokenTypes.U_CHAR, 'U+1');
+      recogniseToken(TokenTypes.U_CHAR, 'U+A');
+      recogniseToken(TokenTypes.U_CHAR, 'U+1B');
+      recogniseToken(TokenTypes.U_CHAR, 'U+1CD');
+      recogniseToken(TokenTypes.U_CHAR, 'U+1EF0');
+      recogniseToken(TokenTypes.U_CHAR, 'U+23456');
+      recogniseToken(TokenTypes.U_CHAR, 'U+789ABC');
+    });
+    it("can recognise a STRING token (single quote)", () => {
+      recogniseToken(TokenTypes.STRING, '\'10.0\'');
+    });
+    it("can recognise a STRING token (double quote)", () => {
+      recogniseToken(TokenTypes.STRING, '"10.0"');
+    });
+    it("can recognise a STRING token (repeated quotes)", () => {
+      recogniseTokens(
+        '\'hello\' \'world\'',
+        [
+          new Token(TokenTypes.STRING,     '\'hello\''),
+          new Token(TokenTypes.WHITESPACE, ' ', 1, 8),
+          new Token(TokenTypes.STRING,     '\'world\'', 1, 9),
+        ]
+      );
+    });
+    it("does not pick out tokens from inside strings", () => {
+      recogniseToken(TokenTypes.STRING, '"store"');
+    });
```

Comment:
```suggestion
    it("does not pick out tokens from inside strings", () => {
      recogniseToken(TokenTypes.STRING, '"store"');
      recogniseToken(TokenTypes.STRING, '"\'"');
      recogniseToken(TokenTypes.STRING, "'\"'");
    });
```

Two of the more popular strings!